### DISCUSSION TITLE
Installable Kedro catalog (Mini-Kedro)

### DISCUSSION THREAD

**Discussion by WaylonWalker** [kedro-org/kedro#4273] [Idea]
Would it make sense to make `mini-kedro` installable?  My use case for projects like that are users doing EDA and just want easy access to the data with no fuss.  

If it is something that makes sense I propose adding a setup.py to make it installable, and a single module that sets the catalog up for them, then they can access the project's data as follows.

``` python
import my_mini_kedro as mmk
mmk.catalog.load('my_dataset')
```

## Alternatively

This could also be a separate starter that is a `kedro-catalog` starter.

**Comment by JavierHernandezMontes** [comment]
@WaylonWalker do you know a way to load a certain dataset of the catalog with the current version of kedro 0.17.0? as you propose to do.

**Comment by WaylonWalker** [comment]
I just added a very basic `setup.py` and an `__init__.py` with a catalog loader to make it work.

**Comment by WaylonWalker** [comment]
@JavierHernandezMontes is your data local or remote?  For local data that needs packaged in it might be a bit trickier.

**Comment by noklam** [comment]
@WaylonWalker Is this still relevant with the `IPython` extension?

**Comment by astrojuanlu** [comment]
At the moment, using Kedro without using the project template is entirely possible: `pip install kedro` and then instantiating the catalog directly:

```
from kedro.config import OmegaConfigLoader
from kedro.io import DataCatalog

conf_loader = OmegaConfigLoader("conf")
conf_catalog = conf_loader.get("catalog")
catalog = DataCatalog.from_config(conf_catalog)

catalog.load(...)
```

On IPython & Jupyter, this is one line:

```
%load_ext kedro.ipython

catalog.load(...)
```

I agree it would be nice to make the boilerplate above go away, so I'm moving this issue and renaming it for our consideration. It will have more visibility on the framework repo.

**Comment by astrojuanlu** [comment]
We are exploring alternative approaches, see gh-2819

**Comment by astrojuanlu** [comment]
And #2967

**Comment by astrojuanlu** [comment]
More evidence of users using the Kedro catalog without the pipelines: https://github.com/kedro-org/kedro/issues/2898#issuecomment-1736000094

**Comment by astrojuanlu** [comment]
Another user asking for exactly this https://linen-slack.kedro.org/t/16593946/is-there-a-way-of-installing-only-the-data-catalog-part-of-k#b6d532c4-2d7f-4add-b0ee-b0bfcffbdd5e

**Comment by astrojuanlu** [comment]
Interesting realisation today. Users have been able to use the `DataCatalog` as a standalone component since basically forever. The `mini-kedro` starter was created in Kedro 0.17.0:

https://github.com/kedro-org/kedro-starters/blob/0.17.0/mini-kedro/%7B%7B%20cookiecutter.repo_name%20%7D%7D/Example%20Notebook.ipynb

Then in 2023 we made a big push to show that you can use Kedro standalone components (#2855, #3128)

And yet, even some of our power users had no idea it's possible to do this.

In #3659 I gave a detailed rationale of the _technical reasons_ why it's just better to have fewer dependencies and enable users to install only the parts they want. But I was missing the _marketing reasons_: old users have preconceived notions about what Kedro can and cannot do, and new users find a wall of documentation explaining the framework way, so most don't realise that there's a library way as well.

**Comment by astrojuanlu** [comment]
This is still important for us.

As part of a issue cleanup, and in line with a new scheme in which we want to use Discussions for feature requests & enhancement proposals #3767, I'm moving this to a discussion. Let's continue there.

**Comment by astrojuanlu** [comment]
Related, new discussion https://github.com/kedro-org/kedro/discussions/4361
